nmpolicy/internal/resolver/resolver_test.go

```diff
@@ -0,0 +1,199 @@
+routes:
+  running:
+  - destination: 0.0.0.0/0
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+  - destination: 1.1.1.0/24
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+  config:
+  - destination: 0.0.0.0/0
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+  - destination: 1.1.1.0/24
+    next-hop-address: 192.168.100.1
+    next-hop-interface: eth1
+    table-id: 254
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up
+    ipv4:
+      address:
+      - ip: 10.244.0.1
+        prefix-length: 24
+      - ip: 169.254.1.0
+        prefix-length: 16
+      dhcp: false
+      enabled: true
+  - name: eth2
+    type: ethernet
+    state: down
+    ipv4:
+      address:
+      - ip: 1.2.3.4
+        prefix-length: 24
+      dhcp: false
+      enabled: false
+`
+var resolver Resolver
+
+func TestResolve(t *testing.T) {
+	createResolver()
+
+	var tests = []struct {
+		astYaml      string
+		expectedYaml string
+		err          string
+	}{
+		{
+			astYaml:`
+pos: 1
+eqfilter:
+- pos: 2
+  identity: currentState
+- pos: 3
+  path:
+  - pos: 4
+    identity: routes
+  - pos: 5
+    identity: running
+  - pos: 6
+    identity: destination
+- pos: 7
+  string: 0.0.0.0/0
+`,
+			expectedYaml: `
+routes:
+ running:
+ - destination: 0.0.0.0/0
+   next-hop-address: 192.168.100.1
+   next-hop-interface: eth1
+   table-id: 254
+`,
+		},
+		{
+			astYaml:`
+pos: 1
+eqfilter:
+- pos: 2
+  identity: currentState
+- pos: 3
+  path:
+  - pos: 4
+    identity: interfaces
+  - pos: 5
+    identity: state
+- pos: 6
+  string: down
+`,
+			expectedYaml: `
+interfaces:
+  - name: eth2
+    type: ethernet
+    state: down
+    ipv4:
+      address:
+      - ip: 1.2.3.4
+        prefix-length: 24
+      dhcp: false
+      enabled: false
+`,
+		},
+		{
+			astYaml:`
+pos: 1
+eqfilter:
+- pos: 2
+  identity: currentState
+- pos: 3
+  path:
+  - pos: 4
+    identity: interfaces
+  - pos: 5
+    identity: ipv4
+  - pos: 6
+    identity: address
+  - pos: 7
+    identity: ip
+- pos: 8
+  string: 10.244.0.1
+`,
+			expectedYaml: `
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up
+    ipv4:
+      address:
+      - ip: 10.244.0.1
+        prefix-length: 24
+      - ip: 169.254.1.0
+        prefix-length: 16
+      dhcp: false
+      enabled: true
+`,
+		},
+		{
+			astYaml:`
+pos: 1
+eqfilter:
+- pos: 2
+  identity: currentState
+- pos: 3
+  path:
+  - pos: 4
+    identity: interfaces
+  - pos: 5
+    identity: ipv4
+  - pos: 6
+    identity: address
+- pos: 7
+  string: 10.244.0.1
+`,
+			err: "invalid path",
+		},
+	}
+
+	for testIndex, currentTest := range tests {
+		t.Run(fmt.Sprintf("%d", testIndex+1), func(t *testing.T) {
+			obtainedAST := &ast.Node{}
+			err := yaml.Unmarshal([]byte(currentTest.astYaml), obtainedAST)
+			if err != nil {
+				panic(fmt.Sprintf("Couldn't unmarshal the ast %s, err: %v", currentTest.astYaml, err))
+			}
```

Comment:
```suggestion
			assert.NoError(t, yaml.Unmarshal([]byte(currentTest.astYaml), obtainedAST))
```

Reply:
I used a panic to be clear it is a bug in the test itself and not in the tested object. But will change to your suggestion.